js_modules/dagit/packages/core/src/assets/AssetPartitions.tsx

```diff
@@ -105,10 +117,31 @@ export const AssetPartitions: React.FC<Props> = ({
       return [];
     }
 
-    const {dimension, selectedRanges} = selections[idx];
-    const allKeys = dimension.partitionKeys;
+    const {selectedRanges} = selections[idx];
+    const allKeys = sortedPartitions[idx];
+    const sort = selectionSorts[idx] || defaultSort(selections[idx].dimension.type);
+
+    function translateIndex(idx: number) {
+      if (sort === -1) {
+        return allKeys.length - idx - 1;
+      }
+      return idx;
+    }
+
+    function sliceTranslated(arr: Array<any>, start: number, end: number) {
+      const index1 = translateIndex(start);
+      const index2 = translateIndex(end);
+      const smaller = index1 > index2 ? index2 : index1;
+      const greater = index1 > index2 ? index1 : index2;
+      return arr.slice(smaller, greater + 1);
+    }
+
     const getSelectionKeys = () =>
-      uniq(selectedRanges.flatMap(([start, end]) => allKeys.slice(start.idx, end.idx + 1)));
+      uniq(
+        selectedRanges.flatMap(([start, end]) => {
+          return sliceTranslated(allKeys, start.idx, end.idx);
+        }),
```

Comment:
Hmm, are there test cases for the  Z->F, C->A case you're talking about? Let me play with this a bit more. I was wary of not memoizing the reverse since I know there could be a very large number of assets

Reply:
Hmm i don't know if there's a test case yet -- it's sorta on my todo list to break this function out of the component and test it as a standalone thing. 

I think you're right that we want to memoize the `.reverse()` call since partitionKeys can be 200k+ items -- I think the question is how often `selections` changes, because it might be that .reverse() on the selections.dimensions.partitionKeys and just reversing the result of this function call roughly the same number of renders 😅

Reply:
<img width="1091" alt="Screen Shot 2023-03-17 at 11 07 07 AM" src="https://user-images.githubusercontent.com/2286579/225943810-a6390cd3-88fd-4f65-b832-90fc6b7c059f.png">

Looks fine to me

Reply:
You're right though I should just reverse it at the end. Not sure why I glossed over that 😅